### PR TITLE
Charts: Added support for patterns

### PR DESCRIPTION
**Pull request by dlabrecq**
Added support for patterns. For examples, please see https://patternfly-react-pr-7390.surge.sh/charts/patterns.

Reviewed with @mceledonia, @heyethankim, & @mcarrano.

Screen Shot 2022-05-15 at 10 32 10 PM

Design issue: https://github.com/patternfly/patternfly-design/issues/1136
Closes: https://github.com/patternfly/patternfly-react/issues/7537

### CODE REVIEW

File: packages/react-charts/src/components/Patterns/examples/patterms.md [hunk 1]
```diff
@@ -0,0 +1,791 @@
...
+## Introduction
+Note: PatternFly React charts live in its own package at [@patternfly/react-charts](https://www.npmjs.com/package/@patternfly/react-charts)!
+
+PatternFly React charts are based on the [Victory](https://formidable.com/open-source/victory/docs/victory-chart/) chart library, along with additional functionality, custom components, and theming for PatternFly. This provides a collection of React based components you can use to build PatternFly patterns with consistent markup, styling, and behavior.
+
+## Examples
+### Basic pie chart
+```js
+import React from 'react';
+import { ChartPie, ChartThemeColor } from '@patternfly/react-charts';
```
Comment (tlabaj) [inline packages/react-charts/src/components/Patterns/examples/patterms.md:48]: ChartThemeColor is not used in this example

File: packages/react-charts/src/components/Patterns/examples/patterms.md [hunk 2]
```diff
@@ -0,0 +1,791 @@
...
+import { 
+  Chart, 
+  ChartArea, 
+  ChartAxis, 
+  ChartGroup, 
+  ChartLegend,
+  ChartLegendTooltip,
+  ChartScatter, 
+  ChartThemeColor,
+  ChartVoronoiContainer,
```
Comment (tlabaj) [inline packages/react-charts/src/components/Patterns/examples/patterms.md:389]: ChartVoronoiContainer is not used in this example.

File: packages/react-charts/src/components/Chart/Chart.tsx [hunk 3]
```diff
@@ -407,6 +436,10 @@ export interface ChartProps extends VictoryChartProps {
    * @deprecated Use PatternFly's pf-theme-dark CSS selector
    */
   themeVariant?: string;
+  /**
+   * Generate default pattern defs and populate patternScale
+   */
+  usePatternDefs?: boolean;
```
Comment (tlabaj) [inline packages/react-charts/src/components/Chart/Chart.tsx:442]: nit.  Typically in patternfly-core we prepend out booleans with `is` or sometimes `has`

File: packages/react-charts/src/components/ChartUtils/chart-patterns.tsx [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+  patternScale?: string[];
+  usePatternDefs?: boolean;
+}
+
+/**
+ * Patterns were pulled from v3.0.3 of the script below, which uses the MIT license.
+ * See https://github.com/highcharts/pattern-fill/blob/master/pattern-fill-v2.js
+ * @private
+ */
+const patterns: any = [
```
[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartUtils/chart-patterns.tsx:18]: Do you mean add design docs? @heyethankim is working on adding docs regarding when to use patterns, so I'll leave that to him.

File: packages/react-charts/src/components/Chart/Chart.tsx (hunk 3 above)
Comment (dlabrecq) [reply packages/react-charts/src/components/Chart/Chart.tsx]: Changed to `isPatternDefs`

File: packages/react-charts/src/components/Patterns/examples/patterms.md (hunk 2 above)
Comment (dlabrecq) [reply packages/react-charts/src/components/Patterns/examples/patterms.md]: Done

File: packages/react-charts/src/components/Patterns/examples/patterms.md (hunk 1 above)
Comment (dlabrecq) [reply packages/react-charts/src/components/Patterns/examples/patterms.md]: Done

**Review by tlabaj — APPROVED** [review]
lGTM

**Comment by mcarrano** [comment]
@dlabrecq I'm taking another look at this and just had a couple of questions. Can you explain again how the patterns are generated/assigned? Are these just mapped by default based on the color theme chosen? Is it possible for the consumer using this to explicitly select the pattern that will be applied to a specific color, and if yes, how would they do that?

**Review by nicolethoen — APPROVED** [review]

**Comment by dlabrecq** [comment]
The patterns are dynamically generated, based on the selected color theme or custom color scale. Patterns are applied in a specific order, similar to the chart's multi-color theme defined by PatternFly. However, devs can create their own custom patterns.

**Comment by dlabrecq** [comment]
@mcarrano Patterns are dynamically generated, based on the selected color theme or custom color scale. Then patterns are applied in a specific order, similar to the chart color themes defined by PatternFly. If you were to use the multi-color theme; for example, colorized patterns would be output in that exact order. However, devs can create their own custom patterns.

**Comment by dlabrecq** [comment]
Added beta flag per @mcarrano

**Review by mcarrano — APPROVED** [review]
Thanks @dlabrecq . I'm good with this.

**Review by tlabaj — APPROVED** [review]

**Pull request merged**